src/main/java/org/broadinstitute/hellbender/tools/exome/BayesianHetPulldownCalculator.java

```
@@ -0,0 +1,671 @@
+
+    /* default priors */
+    private static final double DEFAULT_PRIOR_REF_HOM = 0.5; /* a homozygous site being the ref allele */
+    private static final double DEFAULT_PRIOR_HET = 0.5; /* a site being heterozygous */
+
+    /* a third of the minimum sequencing error (for safeguarding log likelihood calculations) */
+    private static final double DEFAULT_MIN_BASE_ERROR_THIRD = 1e-6;
+
+    /* approximate number of status updates printed to log */
+    private static final int NUMBER_OF_SITES_PER_LOGGED_STATUS_UPDATE = 10000;
+
+    /**
+     * Constructor of {@link BayesianHetPulldownCalculator} object
+     *
+     * NOTE: The default {@link HetPriorType} is BALANCED. Make a call to
+     *       {@link BayesianHetPulldownCalculator#useHeterogeneousHetPrior} to switch to the HETEROGENEOUS prior.
+     *
+     * @param refFile the reference genome file
+     * @param snpIntervals {@link IntervalList} of common SNPs
+     * @param minMappingQuality minimum phred mapping quality
+     * @param minBaseQuality minimum phred base quality
+     * @param readDepthThreshold minimum read depth
+     * @param validationStringency validation stringency
+     * @param errorProbabilityAdjustmentFactor (experimental) multiplicative factor for read and mapping error
+     *                                         probabilities
+     */
+    public BayesianHetPulldownCalculator(final File refFile, final IntervalList snpIntervals,
+                                         final int minMappingQuality, final int minBaseQuality,
+                                         final int readDepthThreshold, final ValidationStringency validationStringency,
+                                         final double errorProbabilityAdjustmentFactor) {
+
+        ParamUtils.isPositiveOrZero(minMappingQuality, "Minimum mapping quality must be nonnegative.");
+        ParamUtils.isPositiveOrZero(minBaseQuality, "Minimum base quality must be nonnegative.");
+
+        this.refFile = refFile;
```

Is always better to fail early, that is why.